xio-core/src/main/java/com/xjeffrose/xio/client/ClientConfig.java

```diff
@@ -74,4 +92,78 @@ public static ClientConfig fromConfig(String key, Config config) {
   public static ClientConfig fromConfig(String key) {
     return fromConfig(key, ConfigFactory.load());
   }
+
+  public static ClientConfig.Builder newBuilder(ClientConfig fallbackObject) {
+    return new ClientConfig.Builder(fallbackObject);
+  }
+
+  /**
+   * Used to create a ClientConfig at runtime.
+   *
+   * <p>If a value is not set, it defaults to using the default object's value.
+   */
+  public static class Builder {
+    private ClientConfig fallbackObject;
+    private Map<ChannelOption<Object>, Object> bootstrapOptions;
+    private String name;
+    private TlsConfig tls;
+    private boolean messageLoggerEnabled;
+    private InetSocketAddress local;
+    private InetSocketAddress remote;
+    private IdleTimeoutConfig idleTimeoutConfig;
+
+    private Builder(ClientConfig fallbackObject) {
```

Comment:
This is a ClientConfig instead of a Config so that the user has the capabilities to create a default object once and use it over and over instead of newing up one everytime with the same Config

Reply:
Same reasoning in ProxyRouteConfig as well